### DISCUSSION TITLE
Add an option to enforce all lower-case branch names for cross platform compatibility

### DISCUSSION THREAD

**Discussion by redwyre** [community/community#39010] [Repositories]
### Select Topic Area

Product Feedback

### Body

Hi,

We have a number of programmers, all developing on windows, and we have a standard of creating branches in the form "name/task".
When creating these branches people sometimes change the case of their name part, which randomly causes issues with other developers when syncing because git uses the file system for refs and breaks when there is different case.

Having an option that blocks any branches names that contain uppercase characters from being pushed would eliminate the problem (this is not possible with the current branch naming rules).

In the meantime we are using local scripts to try and block branches from being created but since they need to be manually enabled there's always a chance another one will get through.

**Comment by patrick-knight** [comment]
[Commit metadata ](https://docs.github.com/en/enterprise-cloud@latest/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/creating-rulesets-for-a-repository#using-regular-expressions-for-commit-metadata)rules could be an option for you to limit branch names with a specific regex pattern.

**Comment by Strepto** [comment]
Just so you know I found a way to enforce lower case branches. But I also suggest that github creates a built-in setting for this, as setting up this regex for each and every repository is time consuming and annoying.

See: https://github.com/orgs/community/discussions/20518?sort=top#discussioncomment-9167049
